Examples/Java/Sources/Board.java

```diff
@@ -0,0 +1,56 @@
+//
+// Board.java
+// Autogenerated by plank
+//
+// DO NOT EDIT - EDITS WILL BE OVERWRITTEN
+// @generated
+//
+
+package com.pinterest.models;
+
+import android.support.annotation.IntDef;
+import com.google.auto.value.AutoValue;
+import java.net.URI;
+import java.util.Set;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringDef;
+import java.lang.annotation.Retention;
+import java.util.Date;
+import java.util.List;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Map;
+
+@AutoValue
+public abstract class Board {
+
+
+    public abstract @SerializedName("id") @Nullable String identifier();
+    public abstract @SerializedName("contributors") @Nullable Set<User> contributors();
+    public abstract @SerializedName("counts") @Nullable Map<String, Integer> counts();
+    public abstract @SerializedName("created_at") @Nullable Date createdAt();
+    public abstract @SerializedName("creator") @Nullable Map<String, String> creator();
+    public abstract @SerializedName("description") @Nullable String descriptionText();
+    public abstract @SerializedName("image") Image image();
+    public abstract @SerializedName("name") @Nullable String name();
+    public abstract @SerializedName("url") @Nullable URI url();
+    public static Builder builder() {
+        return new AutoValue_Board.Builder();
+    }
+    abstract Builder toBuilder();
+    @AutoValue.Builder
+    public abstract static class Builder {
+    
+    
+        public abstract Builder setIdentifier(@Nullable String value);
+        public abstract Builder setContributors(@Nullable Set<User> value);
+        public abstract Builder setCounts(@Nullable Map<String, Integer> value);
+        public abstract Builder setCreatedAt(@Nullable Date value);
+        public abstract Builder setCreator(@Nullable Map<String, String> value);
+        public abstract Builder setDescriptionText(@Nullable String value);
+        public abstract Builder setImage(Image value);
+        public abstract Builder setName(@Nullable String value);
+        public abstract Builder setUrl(@Nullable URI value);
+        public abstract Board build();
+    
+    }
```

Comment:
@rahul-malik 

Feature request to add
```java
public static TypeAdapter<Interest> jsonAdapter(Gson gson) {
  return new AutoValue_Interest.GsonTypeAdapter(gson);
}
```
for use with https://github.com/rharter/auto-value-gson
(https://github.com/rharter/auto-value-moshi uses the same format)

This is an AutoValue extension that generates optimized TypeAdapter/JsonAdapter implementations.

The import line for TypeAdapter is `import com.google.gson.TypeAdapter;`

Reply:
Adding now, thanks for the feedback!

Reply:
nice! thanks!